torch/lib/THD/base/data_channels/DataChannelGloo.cpp

```diff
@@ -67,33 +65,44 @@ void DataChannelGloo::RequestGloo::wait() {
 
 DataChannelGloo::DataChannelGloo()
   : _rank(load_rank_env())
+  , _store(nullptr)
+  , _cache(nullptr)
 {
-  _store = std::unique_ptr<Store>(new thd::Store());
-
   if (_rank == 0) {
     _num_processes = load_world_size_env();
-    auto num_proc_str = std::to_string(_num_processes);
-    _store->set("world_size",
-            std::vector<char>(num_proc_str.begin(), num_proc_str.end()));
-  } else {
-    auto world_size = _store->get("world_size");
-    _num_processes = std::atoll(
-            std::string(world_size.begin(), world_size.end()).c_str());
   }
 
-  ::gloo::transport::tcp::attr attr; // default options listen on all interfaces
+  ::gloo::transport::tcp::attr attr("localhost"); // default options listen on host
```

Comment:
Remove this. You can't connect to a server that listens on localhost from a different machine.